docs/reference_guide.md

```diff
@@ -99,7 +99,7 @@ This instruments the tcp_v4_connect() kernel function using a kprobe, with the f
 - ```struct pt_regs *ctx```: Registers and BPF context.
 - ```struct sock *sk```: First argument to tcp_v4_connect().
 
-The first argument is always ```struct pt_regs *```, the remainder are the arguments to the function (they don't need to be specified, if you don't intend to use them).
+The first argument is always ```struct pt_regs *```, the remainder are the arguments to the function (they don't need to be specified, if you don't intend to use them). Limitations: maximum of 4 args including ctx.
```

Comment:
Yes, the printk limitation is indeed as you say (3 parameters). A better error message would be most welcome, the check should be pretty simple. Try putting it into `BTypeVisitor::VisitCallExpr` in the `if (Decl->getName() == "bpf_trace_printk")` section. A test case to go with it in the same style as `tests/python/test_clang.py:test_too_many_args()` would be ideal.